### PR TITLE
Improved documentation, distance() function for vertices

### PR DESCRIPTION
**Pull request by j-ac**
-Added documentation comments
-vertices now have distance()

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -136,20 +138,36 @@ pub struct Coordinate3d {
...
     vertices: Vec<Coordinate3d>,
     origin: Coordinate3d,
 }
 
 impl Mesh {
+    ///Translate a mesh by a vector quantity. This function is implemented by calling
+    ///Coordinate3d::translate for each vertex in the mesh.
     pub fn translate(&mut self, vector: ColumnVector<3>) {
         for vertex in self.vertices.iter_mut() {
             vertex.translate(vector.clone()); //TODO: avoid clone() by implementing + by reference.
```
Comment (will459) [inline src/lib.rs]: translate takes by value, you don't need clone here.

File: src/lib.rs [hunk 2]
```diff
@@ -3,12 +3,13 @@ use std::ops;
 use std::ops::Mul;
 use std::ops::{Index, IndexMut};
 
+///An abstraction over a 2D matrix of f64s. the foundation of this library
```
Comment (will459) [inline src/lib.rs]:
```suggestion
/// An abstraction over a 2D matrix of f64s. the foundation of this library
```
nit, a space is more readable

File: src/lib.rs [hunk 3]
```diff
@@ -3,12 +3,13 @@ use std::ops;
...
 #[derive(Debug, Clone, Copy, PartialEq, PartialOrd)]
 pub struct Matrix<const ROWS: usize, const COLS: usize> {
     matrix: [[f64; COLS]; ROWS],
 }
 
-//Matrix with one column
+///A Matrix with only one column
```
Comment (will459) [inline src/lib.rs]:
```suggestion
/// A [`Matrix`] with only one column
```
You can make link references like this.  This works for std:: items like [\`f64\`] above

**Comment by will459** [comment]
You are missing crate level documentation and Coordinate3d.   Adding #![deny(missing_docs)] or #![warn(missing_docs)] to deny or warn about them.

**Review by will459 — CHANGES_REQUESTED** [review]

File: src/lib.rs [hunk 4]
```diff
@@ -136,20 +138,36 @@ pub struct Coordinate3d {
...
+    fn distance(self, cord: Coordinate3d) -> f64 {
+        //pythagorean theorem
+        //sqrt((x1 + x2)^2 + (y1 + y2)^2 + (z1 + z2)^2)
+        let mut ans: f64 = 0.0;
+        for axis in 0..3 {
+            ans += (self.pos[(axis, 1)] + cord.pos[(axis, 1)]).powi(2);
+        }
+
+        ans = ans.sqrt();
+        ans
```
Comment (ben-j-c) [inline src/lib.rs]: Can't you just `ans.sqrt()` here?

**Review by ben-j-c — CHANGES_REQUESTED** [review]

File: src/lib.rs (hunk 4 above)
Comment (j-ac) [reply src/lib.rs]: Fixed just a second ago in today's commit to PR_1

File: src/lib.rs [hunk 5]
```diff
@@ -131,28 +138,44 @@ impl<const ROWS: usize, const COLS: usize> ops::Mul<f64> for Matrix<ROWS, COLS>
...
+    /// Calculates the distance between two `Coordinate3d`'s
+    fn distance(self, cord: Coordinate3d) -> f64 {
+        //pythagorean theorem
+        //sqrt((x1 + x2)^2 + (y1 + y2)^2 + (z1 + z2)^2)
+        let mut ans: f64 = 0.0;
+        for axis in 0..3 {
+            ans += (self.pos[(axis, 1)] + cord.pos[(axis, 1)]).powi(2);
+        }
+
+        ans.sqrt()
```
Comment (ben-j-c) [inline src/lib.rs]: Misnomer. Should be sum or something like that.  :)

Comment (j-ac) [reply src/lib.rs]: Sure. I'll fix

**Review by will459 — APPROVED** [review]

**Pull request merged**